extensions/classifiers/LDAStringClassifier/LDAStringClassifier.py

```diff
@@ -597,3 +598,90 @@ def predict(self, predicting_data):
         """
         doc_ids = self._add_docs_for_predicting(predicting_data)
         return [self._predict_label_for_doc(doc_id) for doc_id in doc_ids]
+
+    def validate(self, classifier_data):
+        """Validates cached classifier data.
+
+        Args:
+            classifier_data: dict. Consists of various parameter of the
+                classifier.
+        """
+
+        float_properties = [
+            '_alpha',
+            '_beta',
+            '_prediction_threshold'
+        ]
+        int_properties = [
+            '_training_iterations',
+            '_prediction_iterations',
+            '_num_labels',
+            '_num_docs',
+            '_num_words'
+        ]
+        dict_properties = [
+            '_label_to_id',
+            '_word_to_id'
+        ]
+        list_properties = [
+            '_w_dp',
+            '_b_dl',
+            '_l_dp',
+            '_c_dl',
+            '_c_lw',
+            '_c_l'
+        ]
+
+        for float_property in float_properties:
+            if float_property not in classifier_data:
+                raise utils.ValidationError(
+                    'Expected %s to be a key in classifier_data' %
+                    float_property)
+            if not isinstance(classifier_data[float_property], float):
+                raise utils.ValidationError(
+                    ('Expected %s to be a float, received %s' % (
```

Comment:
Here and below: I think you can remove the outer parentheses (i.e. the one coming at the start of this line and its matching counterpart).